File: leetcode/SingleNumber/singleNumber.java
public class Solution {
    public int singleNumber(int[] A) {
        // Note: The Solution object is instantiated only once and is reused by each test case.
        int ans = A[0];
        for (int i = 1; i < A.length; i++) ans ^= A[i];
        return ans;
    }
}

File: leetcode/SingleNumberIII/singleNumber.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        int diff = 0;
        for (auto x : nums) diff ^= x;
        diff &= (-diff);
        vector<int> ans = {0, 0};
        for (auto x : nums) {
            if (x & diff) ans[0] ^= x;
            else ans[1] ^= x;
        }
        return ans;
    }
};
